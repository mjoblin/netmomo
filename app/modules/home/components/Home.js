import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { bindActionCreators } from 'redux';

import { ConnectionStatus } from 'AppRoot/components';
import appModule from 'AppRoot/modules/app';
import settingsModule from 'AppRoot/modules/settings';
import './style.scss';


export const Home = () => (
    <div className="home">
        <h1>Welcome to netmomo</h1>

        netmomo is a <a href="https://netdumplings.readthedocs.org">NetDumplings</a> dumpling
        eater. It shows you the contents of the dumplings collected from your network.

        <p />
        <div className="shifty-connection-status">
            Dumpling hub connection status: <ConnectionStatus />
        </div>
        <p />

        If you cannot connect to the dumpling hub then try changing
        the <NavLink to='/settings'>Settings.</NavLink>
    </div>
);

Home.propTypes = {
    shiftyConnected: PropTypes.bool.isRequired,
    settings: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
    shiftyConnected: appModule.selectors.shiftyConnected(state),
    settings: settingsModule.selectors.getSettings(state),
});

const mapDispatchToProps = dispatch => {
    return {
        actions: bindActionCreators({
            shiftyConnect: appModule.actions.shiftyConnect,
            shiftyDisconnect: appModule.actions.shiftyDisconnect
        }, dispatch)
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Home);
