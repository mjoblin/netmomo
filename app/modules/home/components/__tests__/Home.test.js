import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../Home';

describe('Home', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <Home
                shiftyConnected={false}
                settings={{}}
            />
        );

        expect(component).toMatchSnapshot();
    });
});