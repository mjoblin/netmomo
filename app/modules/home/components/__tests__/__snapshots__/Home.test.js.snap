// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    settings={Object {}}
    shiftyConnected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome to netmomo
        </h1>,
        "netmomo is a ",
        <a
          href="https://netdumplings.readthedocs.org"
        >
          NetDumplings
        </a>,
        " dumpling eater. It shows you the contents of the dumplings collected from your network.",
        <p />,
        <div
          className="shifty-connection-status"
        >
          Dumpling hub connection status: 
          <Connect(ConnectionStatus) />
        </div>,
        <p />,
        "If you cannot connect to the dumpling hub then try changing the ",
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/settings"
        >
          Settings.
        </NavLink>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to netmomo",
        },
        "ref": null,
        "rendered": "Welcome to netmomo",
        "type": "h1",
      },
      "netmomo is a ",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "NetDumplings",
          "href": "https://netdumplings.readthedocs.org",
        },
        "ref": null,
        "rendered": "NetDumplings",
        "type": "a",
      },
      " dumpling eater. It shows you the contents of the dumplings collected from your network.",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Dumpling hub connection status: ",
            <Connect(ConnectionStatus) />,
          ],
          "className": "shifty-connection-status",
        },
        "ref": null,
        "rendered": Array [
          "Dumpling hub connection status: ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      "If you cannot connect to the dumpling hub then try changing the ",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Settings.",
          "to": "/settings",
        },
        "ref": null,
        "rendered": "Settings.",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to netmomo
          </h1>,
          "netmomo is a ",
          <a
            href="https://netdumplings.readthedocs.org"
          >
            NetDumplings
          </a>,
          " dumpling eater. It shows you the contents of the dumplings collected from your network.",
          <p />,
          <div
            className="shifty-connection-status"
          >
            Dumpling hub connection status: 
            <Connect(ConnectionStatus) />
          </div>,
          <p />,
          "If you cannot connect to the dumpling hub then try changing the ",
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/settings"
          >
            Settings.
          </NavLink>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to netmomo",
          },
          "ref": null,
          "rendered": "Welcome to netmomo",
          "type": "h1",
        },
        "netmomo is a ",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "NetDumplings",
            "href": "https://netdumplings.readthedocs.org",
          },
          "ref": null,
          "rendered": "NetDumplings",
          "type": "a",
        },
        " dumpling eater. It shows you the contents of the dumplings collected from your network.",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Dumpling hub connection status: ",
              <Connect(ConnectionStatus) />,
            ],
            "className": "shifty-connection-status",
          },
          "ref": null,
          "rendered": Array [
            "Dumpling hub connection status: ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "If you cannot connect to the dumpling hub then try changing the ",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Settings.",
            "to": "/settings",
          },
          "ref": null,
          "rendered": "Settings.",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
